sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -26,9 +26,12 @@ trait FunctionRegistry {
 
   def registerFunction(name: String, builder: FunctionBuilder): Unit
 
-  def lookupFunction(name: String, children: Seq[Expression]): Expression
-
   def caseSensitive: Boolean
+
+  def lookupFunction(
+      name: String,
+      children: Seq[Expression],
+      distinct: Boolean = false): Expression
```

Comment:
Hmm, I have not a smart idea.
Does you know a similar logic to look up user-defined functions in hive?